cmake/BuildBlockFactory.cmake

```diff
@@ -0,0 +1,12 @@
+# Copyright (C) 2019 iCub Facility, Istituto Italiano di Tecnologia
+# Authors: Diego Ferigo <diego.ferigo@iit.it>
+# CopyPolicy: Released under the terms of the LGPLv2.1 or later, see LGPL.TXT
+
+include(YCMEPHelper)
+
+ycm_ep_helper(BlockFactory TYPE GIT
+              STYLE GITHUB
+              REPOSITORY robotology/blockfactory.git
+              TAG master
+              COMPONENT dynamics
+              FOLDER robotology)
```

Comment:
Can you enforce the use of `Matlab` if `ROBOTOLOGY_USES_MATLAB` (or `ROBOTOLOGY_USES_SIMULINK` if if is the case), by adding the 
~~~
CMAKE_ARGS -DUSES_MATLAB:BOOL=${ROBOTOLOGY_USES_MATLAB}
~~~ 
line? See for example in https://github.com/robotology/robotology-superbuild/blob/b11088d69a68b262a6caa3a1de4b0b9d512af832/cmake/Buildyarp-matlab-bindings.cmake#L16 .

The main idea is that in this way from the superbuild one has full control of whether the subprojects in the superbuild compile or not the dependency-related part. Without this, the following problem typically arise (if I search, I probably can find the corresponding issue): 
* A user first compile the superbuild with `ROBOTOLOGY_USES_MATLAB` set to `FALSE`, and without MATLAB installed on his computer (or installed, but it cannot be found by CMake for any reason). Then, it installs MATLAB, and set `ROBOTOLOGY_USES_MATLAB` to `ON`. Unless the `USES_MATLAB` option is explicitly set every time (not just as an initial cache!), the blockfactory will not compile the matlab/simulink part, and the user will be puzzled on why he can't get the controllers from whole-body-controllers to work.